tests/testlib.bsh

```diff
@@ -375,13 +370,26 @@ _begin_common_test ()
     ttouch "${track_touched_file}"
   fi
 
-  if [ "${TESTLIB_SHOW_TIMING-0}" = "1" ]; then
-    _time_0=$(get_time_seconds)
-  fi
-
   if [ "${TESTLIB_RUN_SINGLE_TEST+set}" = "set" ] && \
      [ "$1" != "${TESTLIB_RUN_SINGLE_TEST}" ]; then
     skip_next_test
+  elif [ -n "${TESTLIB_SKIP_TESTS+set}" ] && [[ ${1} =~ ${TESTLIB_SKIP_TESTS} ]]; then
+    skip_next_test
+  fi
+
+  # Run setup if this is the first test
+  if [ "${__testlib_skip_test-}" != "1" ] && \
+     [ "${__testlib_ran_a_test-}" != "1" ] && \
+     type -t setup &> /dev/null && \
+     [ "$(command -v setup)" == "setup" ]; then
+    local __testlib_setup_started=0
+    setup
+    __testlib_setup_started=1
+    __testlib_ran_a_test=1
```

Comment:
I don't see why you need `__testlib_setup_started`. It seems like it's job could be done by `__testlib_ran_a_test`